pupil_src/shared_modules/calibration_routines/fingertip_calibration.py

```diff
@@ -0,0 +1,139 @@
+        super().init_ui()
+        self.menu.label = 'Fingertip Calibration'
+        self.menu.append(Info_Text('Calibrate gaze parameters using finger'))
+
+    def start(self):
+        super().start()
+        audio.say('Starting {}'.format(self.mode_pretty))
+        logger.info('Starting {}'.format(self.mode_pretty))
+        self.active = True
+        self.ref_list = []
+        self.pupil_list = []
+        glfwSetKeyCallback(self.g_pool.main_window, self.on_window_key)
+        self.button.status_text = ''
+        if self.mode == 'calibration':
+            self.press_key = -1
+        elif self.mode == 'accuracy_test':
+            self.fingertip_tracker.train_done = 3
+
+    def stop(self):
+        self.button.status_text = ''
+        audio.say('Stopping  {}'.format(self.mode_pretty))
+        logger.info('Stopping  {}'.format(self.mode_pretty))
+        self.active = False
+        # self.close_window()
+        if self.mode == 'calibration':
+            finish_calibration(self.g_pool, self.pupil_list, self.ref_list)
+        elif self.mode == 'accuracy_test':
+            self.finish_accuracy_test(self.pupil_list, self.ref_list)
+        super().stop()
+
+    def recent_events(self, events):
+        """
+        gets called once every frame.
+        reference positon need to be published to shared_pos
+        if no reference was found, publish 0,0
+        """
+        frame = events.get('frame')
+
+        if self.active and frame:
+            recent_pupil_positions = events['pupil_positions']
+            # Update the marker
+            if self.mode == 'calibration':
+                if self.press_key in (-1, 2):
+                    self.button.status_text = 'Cover the rectangles with your index finger and then press space'
+                elif self.press_key == 1 and self.fingertip_tracker.train_done == 1:
+                    self.button.status_text = 'Press space to start calibration'
+                elif self.press_key == 1 and self.fingertip_tracker.train_done == 2:
+                    self.button.status_text = 'Press F to re-detect the skin color if needed; Press C to stop calibration'
+
+            self.fingertip = self.fingertip_tracker.update(frame.bgr, self.press_key)
+            self.press_key = 0
+
+            if self.fingertip is not None:
+                # Set the pos to be the center of the first detected marker
+                marker_pos = self.fingertip['screen_pos']
+                self.pos = self.fingertip['norm_pos']
+                ref = {}
+                ref['norm_pos'] = self.pos
+                ref['screen_pos'] = marker_pos
+                ref['timestamp'] = frame.timestamp
+                self.ref_list.append(ref)
+            else:
+                self.pos = None  # indicate that no reference is detected
+
+            # Always save pupil positions
+            for p_pt in recent_pupil_positions:
+                if p_pt['confidence'] > self.pupil_confidence_threshold:
```

Comment:
this is no longer required in calibration routines. Please remove the confidence threshold check.